src/server/services/input-types.js

```diff
@@ -3,6 +3,12 @@ const _ = require('lodash');
 const blocks2js = require('./blocks2js');
 const Projects = require('../storage/projects');
 
+let typeTape;
+function withTypeTape(fn) {
+    typeTape = [];
+    const result = fn();
```

Comment:
Do we not need to set `typeTape` back to the previous value before return (like RAII)?